hack/cosign-sign-blob-fulcio.sh

```diff
@@ -22,94 +22,68 @@ set -o errexit
 set -o pipefail
 set -o nounset
 set -o errtrace
+# Enable debugging
+#set -o xtrace
 
 if (( $# != 1 )); then
     echo "usage $0 <file to sign>"
     exit 1
 fi
 
+# We call `cosign initialize` below which will modify the local
+# $HOME/.sigstore/root, so we keep a backup of it and restore it afterwards
+SIGSTORE_ROOT="${HOME}"/.sigstore/root
+SIGSTORE_ROOT_BACKUP="$(mktemp --directory --tmpdir)"
+
+SERVICE_ACCOUNT="${2:-default}"
+
 LOGS="$(mktemp --tmpdir)"
 SIGNATURE="$(mktemp --tmpdir)"
 CERTIFICATE="$(mktemp --tmpdir)"
 TLOG_ENTRY="$(mktemp --tmpdir)"
 cleanup() {
+    [ -o xtrace ] && echo -e "📜 \033[1mCosign logs\033[0m" && cat "${LOGS}"
+    rm -rf "${SIGSTORE_ROOT}"
+    if [ "$(ls -A "${SIGSTORE_ROOT_BACKUP}")" ]; then
+        mkdir -p "${SIGSTORE_ROOT}"
+        mv "${SIGSTORE_ROOT_BACKUP}"/* "${SIGSTORE_ROOT}"
+    fi
+
     rm -f "${LOGS}" "${SIGNATURE}" "${CERTIFICATE}" "${TLOG_ENTRY}"
 }
 trap cleanup exit
 
-kubectl get secret ctlog-public-key > /dev/null 2>&1 || kubectl -n ctlog-system get secret ctlog-public-key -o yaml | yq ".metadata.namespace |= \"$(kubectl get sa -o=jsonpath='{.items[0]..metadata.namespace}')\"" | kubectl apply -f -
+mv -f "${SIGSTORE_ROOT}"/* "${SIGSTORE_ROOT_BACKUP}" > /dev/null 2>&1 || true
 
-kubectl run sign-blob \
-    --quiet \
-    --rm \
-    --stdin \
-    --attach \
-    --image=gcr.io/projectsigstore/cosign \
-    --env=SSL_CERT_FILE=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt \
-    --env=SIGSTORE_CT_LOG_PUBLIC_KEY_FILE=/ctlog/public \
-    --override-type=strategic \
-    --overrides='{
-        "apiVersion": "v1",
-        "spec": {
-            "containers":[
-                {
-                    "name": "sign-blob",
-                    "volumeMounts": [
-                        {
-                            "name": "ctlog-public-key",
-                            "mountPath": "/ctlog",
-                            "secretName": "ctlog-public-key"
-                        }
-                    ]
-                }
-            ],
-            "volumes": [
-                {
-                    "secret": {
-                        "defaultMode": 420,
-                        "name": "ctlog-public-key",
-                        "secretName": "ctlog-public-key"
-                    },
-                    "name": "ctlog-public-key"
-                }
-            ]
-        }
-    }' \
-    -- \
-    sign-blob \
+cosign initialize --mirror http://tuf.localhost --root http://tuf.localhost/root.json
+
+cosign sign-blob \
     --oidc-issuer https://kubernetes.default.svc.cluster.local \
-    --fulcio-url=http://fulcio-server.fulcio-system.svc.cluster.local \
-    --identity-token /var/run/secrets/kubernetes.io/serviceaccount/token \
-    --rekor-url http://rekor-server.rekor-system.svc.cluster.local \
+    --fulcio-url http://fulcio.localhost \
```

Comment:
How is `fulcio.localhost` resolved?

Reply:
You can resolve anything.localhost to the loopback interface (see https://datatracker.ietf.org/doc/html/rfc6761#section-6.3)